bot/roles.go

```diff
@@ -25,7 +28,10 @@ func roleJoinCmd(ctx CommandCtx) error {
 			return fmt.Errorf("error adding user to role: %w", err)
 		} else {
 			log.Printf("Added user with id: %s (%s) to %s role", msg.Author.ID, msg.Author.Username, roleID)
-			ctx.Reply(fmt.Sprintf("You've been added to the group %s.", args[0]))
+			err := ctx.Reply(fmt.Sprintf("You've been added to the group %s.", args[0]))
```

Comment:
Shouldn't most of these `if err != nil { log.Printf(...) }` statements just become `if err != nil { return err }`? You can then handle them and log them (with the command name / args / etc if you want) inside the actual CommandRegistry function.